src/generate-query.ts

```diff
@@ -707,32 +709,24 @@ function getListSizeDirectiveDefaultValue(
 
   const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
   if (
-    !requireOneSlicingArg ||
-    (requireOneSlicingArg.value as BooleanValueNode).value === true
+    requireOneSlicingArg &&
+    (requireOneSlicingArg.value as BooleanValueNode).value === false
   ) {
-    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
+    throw new Error(
```

Comment:
I don't think it is a good idea to throw an error when this is set to false.

Reply:
should it be an `assert` instead then? Since it would be an internal programming error to get to this function otherwise?

Reply:
I don't think we should break when we are using `@listSize` directive in a very ordinary way, but what I'm really asking is if it is possible to address this problem with what I discussed below.